polyply/src/backmap.py

```diff
@@ -173,9 +173,18 @@ def _place_init_coords(self, meta_molecule):
                 cg_coord = meta_molecule.nodes[node]["position"]
                 resid = meta_molecule.nodes[node]["resid"]
                 high_res_atoms = meta_molecule.nodes[node]["graph"].nodes
-                template = orient_template(meta_molecule, node,
-                                           meta_molecule.templates[resname],
-                                           built_nodes)
+
+                try:
+                    template = orient_template(meta_molecule, node,
+                                               meta_molecule.templates[resname],
+                                               built_nodes)
+                except KeyError as err:
+                    msg = ("Couldn't find atom {} of residue {} in the template coordinates. "
+                           "This usually means that you have two or more residues with the same name, "
+                           "but a different number of atoms. All residue names need to "
+                           "describe the same residue.")
+                    raise KeyError(msg.format(err, resname))
```

Comment:
Shouldn't this be done when parsing instead?

Reply:
In principle yes but then you'll have to check each residue in the system against all residues of the same name. For a large system this becomes prohibitively slow, so I don't want to go there. This is really more a last resort safeguard.

Reply:
`if resname in forcefield.blocks: raise KeyError`? You won't even have to check whether the blocks are the same, all you need to know is that the name is not unique.

Reply:
If adjusted this error message leveraging the fact that we have a complete residue graph for molecule. So the comparison can be done on the small residue graphs before going through the structure generation process.